server/graphql_schemas/interest/schema.py

```diff
@@ -20,27 +20,30 @@ class Meta:
 class JoinCategory(relay.ClientIDMutation):
     """Join a category"""
     class Input:
-        category_id = graphene.ID(required=True)
+        categories = graphene.List(graphene.ID)
 
-    joined_category = graphene.Field(InterestNode)
+    joined_category_list = graphene.List(InterestNode)
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
-        category_id = input.get('category_id')
+        categoryIdList = [category for category in input.pop('categories')]
         user = AndelaUserProfile.objects.get(user=info.context.user)
-        user_category = Category.objects.get(pk=from_global_id(category_id)[1])
+        user_category_list = [Category.objects.get(pk=categoryId) for categoryId in categoryIdList]
         try:
-            joined_category = Interest(
-                follower=user,
-                follower_category=user_category
-            )
-            joined_category.save()
+            joined_category_list = []
+            for user_category in user_category_list:
+                joined_category = Interest(
```

Comment:
Great job on this implementation @ipheghe. I think we can make this section of the code more performant. Instead of making database calls for each iteration of the loop which is quite expensive, we can create multiple instances of `Interest` within the loop and then outside the loop, we use the `bulk_create()` method to insert all the records into the DB in a single transaction. We can try:
```python
try:
  joined_category_list = []
  for user_category in user_category_list:
      joined_category  = Interest(follower=user, follower_category=user_category)
      joined_category_list.append(joined_category)
  Interest.objects.bulk_create(joined_category_list)